source/mqtt.c

```diff
@@ -147,10 +149,28 @@ void aws_mqtt_load_error_strings() {
         /* clang-format on */
 #undef AWS_DEFINE_ERROR_INFO_MQTT
 
-        static struct aws_error_info_list s_list = {
+        static struct aws_error_info_list s_error_list = {
             .error_list = s_errors,
             .count = AWS_ARRAY_SIZE(s_errors),
         };
-        aws_register_error_info(&s_list);
+        aws_register_error_info(&s_error_list);
+
+        /* clang-format off */
+        static struct aws_log_subject_info s_logging_subjects[] = {
+            DEFINE_LOG_SUBJECT_INFO(AWS_LS_MQTT_GENERAL, "mqtt", "Misc MQTT logging"),
+            DEFINE_LOG_SUBJECT_INFO(AWS_LS_MQTT_CLIENT, "mqtt-client", "MQTT client and connections"),
+            DEFINE_LOG_SUBJECT_INFO(AWS_LS_MQTT_TOPIC_TREE, "mqtt-topic-tree", "MQTT subscription tree"),
+        };
+        /* clang-format on */
+
+        static struct aws_log_subject_info_list s_logging_subjects_list = {
+            .subject_list = s_logging_subjects,
+            .count = AWS_ARRAY_SIZE(s_logging_subjects),
+        };
+        aws_register_log_subject_info_list(&s_logging_subjects_list);
     }
 }
+
+void aws_mqtt_load_error_strings(void) {
```

Comment:
This should be the other way around, and we should fix the downstream libs

Reply:
What? I followed the HTTP convention, is that wrong?

Reply:
load_error_strings shouldn't be the entry point, library_init should. If HTTP is doing that, it's backwards and we should fix it. @graebm ?

Reply:
Nuked load_error_strings per mike